src/ui/component/publishName/view.jsx

```diff
@@ -0,0 +1,124 @@
+// @flow
+import { CHANNEL_NEW, CHANNEL_ANONYMOUS, MINIMUM_PUBLISH_BID } from 'constants/claim';
+import React, { useState, useEffect } from 'react';
+import { isNameValid } from 'lbry-redux';
+import { FormField } from 'component/common/form';
+import NameHelpText from './name-help-text';
+import BidHelpText from './bid-help-text';
+
+type Props = {
+  name: string,
+  channel: string,
+  uri: string,
+  bid: string,
+  balance: number,
+  isStillEditing: boolean,
+  myClaimForUri: ?StreamClaim,
+  isResolvingUri: boolean,
+  amountNeededForTakeover: number,
+  prepareEdit: ({}, string) => void,
+  updatePublishForm: ({}) => void,
+};
+
+function PublishText(props: Props) {
+  const {
+    name,
+    channel,
+    uri,
+    isStillEditing,
+    myClaimForUri,
+    bid: bidString,
+    isResolvingUri,
+    amountNeededForTakeover,
+    prepareEdit,
+    updatePublishForm,
+    balance,
+  } = props;
+  const [nameError, setNameError] = useState(undefined);
+  const [bidError, setBidError] = useState(undefined);
+  const previousBidAmount = myClaimForUri && Number(myClaimForUri.amount);
+  const bid = Number(bidString);
+
+  function editExistingClaim() {
+    if (myClaimForUri) {
+      prepareEdit(myClaimForUri, uri);
+    }
+  }
+
+  useEffect(() => {
+    let nameError;
+    if (!name) {
+      nameError = __('A name is required');
+    } else if (!isNameValid(name, false)) {
+      nameError = __('LBRY names cannot contain that symbol ($, #, @)');
```

Comment:
Some symbols are missing here. It's probably worth adding a `RESERVED_CHARACTER` const to `lbryUri.js`.